src/runtime/manifest.ts

```diff
@@ -499,12 +499,33 @@ ${e.message}
       }));
 
       const processItems = async (kind: string, f: Function) => {
-        for (const item of items) {
-          if (item.kind === kind) {
-            Manifest._augmentAstWithTypes(manifest, item);
-            await f(item);  // TODO(cypher1): Use Promise.all here.
+        // The items to process may refer to items defined later on. We should do a topological
```

Comment:
A topological sort shouldn't be necessary, we should just process the type related nodes first at this point, having a sort / resolution based algorithm should be overkill for even the medium future.